File: packages/cloud-adapters/observability/aws.ts
import type { ObservabilityAdapter } from "./interface";

const notImpl = (op: string): never => {
  throw new Error(`NotImplemented: cloud-adapters/observability/aws.${op} — CloudWatch+X-Ray impl lands in the AWS-cloud milestone`);
};

export const aws: ObservabilityAdapter = {
  pushLogs:    async () => notImpl("pushLogs"),
  pushMetrics: async () => notImpl("pushMetrics"),
  pushTraces:  async () => notImpl("pushTraces"),
};

File: packages/cloud-adapters/observability/aws.go
package observability

import (
	"context"
	"errors"
)

// AwsAdapter is the CloudWatch+X-Ray-backed observability adapter. Stubbed pending the AWS-cloud milestone.
type AwsAdapter struct{}

func NewAwsAdapter() *AwsAdapter { return &AwsAdapter{} }

func (*AwsAdapter) PushLogs(context.Context, []LogRecord) error {
	return errors.New("NotImplemented: cloud-adapters/observability/aws.PushLogs — CloudWatch+X-Ray impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) PushMetrics(context.Context, []MetricPoint) error {
	return errors.New("NotImplemented: cloud-adapters/observability/aws.PushMetrics — CloudWatch+X-Ray impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) PushTraces(context.Context, []SpanRecord) error {
	return errors.New("NotImplemented: cloud-adapters/observability/aws.PushTraces — CloudWatch+X-Ray impl lands in the AWS-cloud milestone")
}
